### ISSUE TITLE
Relocate single node to directly after depot

### ISSUE THREAD

**Issue by wouterkool** [N-Wouda/PyVRP#142]
#### Describe the bug

In `Exchange<1,0>`, when `prev(V) == depot`, the move `(U,V)` should move node U directly after the depot (before V) if this gives an improvement: `[0, V, U, ...] => [0, U, V, ...]`. The commit https://github.com/N-Wouda/PyVRP/commit/c8ac485c75f27e11b2cfecbbc865d867f9bc6cad defines a test case that fails in current main since it does not find this improved solution even though it exists and the only move in the neighbourhood is this move.

#### Code sample

See https://github.com/N-Wouda/PyVRP/commit/c8ac485c75f27e11b2cfecbbc865d867f9bc6cad.

<details>
INSTALLED VERSIONS
------------------
     pyvrp: 0.0.1
     numpy: 1.24.2
matplotlib: 3.7.0
    vrplib: 1.0.1
      tqdm: 4.64.1
     tomli: 2.0.1
    Python: 3.10.6

</details>

#### Expected Output

The node should be relocated to directly after the depot if this is a strict improvement, so the test case in https://github.com/N-Wouda/PyVRP/commit/c8ac485c75f27e11b2cfecbbc865d867f9bc6cad should succeed.

**Comment by N-Wouda** [comment]
Normal relocate works fine (the first half of the test case), after fixing the test case to return the improved individual. So this:
```python
    # First test normal relocate (325 after 533)
    neighbours = [[] for _ in range(data.num_clients + 1)]
    neighbours[325].append(533)
    ls = LocalSearch(data, pm, rng, neighbours)

    op = Exchange10(data, pm)
    ls.add_node_operator(op)

    indiv = Individual(data, pm, sol_before)
    assert_(indiv_expected.cost() < indiv.cost())
    improved = ls.search(indiv)

    assert_equal(improved, indiv_expected)
```
is OK.

**Comment by N-Wouda** [comment]
https://github.com/N-Wouda/PyVRP/blob/44ebcbb9c7e0a03439a2b05ab094261110000010/pyvrp/cpp/educate/Exchange.h#L56-L63

This has a subtle bug: when `V` is the depot (`V->position == 0`) and we are in a relocate operator (where `M == 0`), the right-hand side of the second condition states `0 + 0 - 1`. In unsigned arithmetic, that wraps around to some large number. This causes the entire function to return `true`, and thus moves wanting to relocate directly after the depot never get evaluated.

I have a fix and will commit that on #143 in a minute.
